app/controllers/merchant/discounts_controller.rb

```diff
@@ -4,4 +4,31 @@ def index
     @discounts = current_user.merchant.discounts.order(:number_items)
   end
 
+  def new
+    @discount = Discount.new
+  end
+
+  def create 
+    @discount = current_user.merchant.discounts.create(discount_params)
+    create_discount(@discount)
+  end
+
+  private
+
+  def discount_params
+    params.permit(:name, :number_items, :percent)
+  end
+
+  def create_discount(discount)
+     if discount.save
+      flash[:notice] = 'Your new discount was saved'
+      session[:failed_discount] = nil
```

Comment:
the session is used to repopulate the form if the creation of the `discount` was not successful